### PR TITLE
Add REST archetype

### PR DESCRIPTION
**Pull request by pith**
Add the REST archetype. But this is not finished yet. I still get an exception when the application starts.

```
com.sun.jersey.api.container.ContainerException: The ResourceConfig instance does not contain any root resource classes.
```

### CODE REVIEW

**Comment by adrienlauer** [comment]
Jersey requires that you exposes at least one REST resource or it fails to start. That's too bad, but we provided a trivial HelloResource in the Web archetype. Maybe we could do this as well, until the rest support itself provides the JSON home resource by default.

File: rest-archetype/src/main/resources/archetype-resources/rest/src/main/java/HelloResource.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package ${package};
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("hello")
```
Comment (pith) [inline rest-archetype/src/main/resources/archetype-resources/rest/src/main/java/HelloResource.java:8]: There is also an HelloResource in this rest-archetype. That's why I don't see why it fails.

**Comment by adrienlauer** [comment]
I think I know. If you generate a project from the archetype that is not in the org.seedstack package the resource is not detected. We should add a seed-bootstrap.properties file in the right place with the generated package.

I think we may have the same problem in the web archetype but it didn't show up for some reason. Can you also add an adequate seed-bootstrap.properties in the app module of the web archetype ?

**Comment by pith** [comment]
You are right good catch. I will add one.

**Comment by pith** [comment]
By the way, I think that the JSON-HOME resource should be exposed on "/". Otherwise when you access the rest archetype on "/" you get the webapp directory.

**Comment by adrienlauer** [comment]
I agree with that. I will look into exposing static and dynamic (REST) resources on "/" with a predefined precedence in case of collision (which should be very rare since static resource almost always have an extension whereas dynamic resource don't and media-type negotiation can further disambiguate the resources).

**Pull request merged**